config/config_test.go

```diff
@@ -426,3 +426,128 @@ func TestNopProvider_Get(t *testing.T) {
 	assert.True(t, v.HasValue())
 	assert.Nil(t, v.Value())
 }
+
+func TestPointerIntField(t *testing.T) {
+	t.Parallel()
+
+	type pointerFieldStruct struct {
+		Name  string
+		Value *int
+	}
+
+	var ptrYaml = `
+ps:
+  name: Hello
+  value: 123
+`
+	p := NewYAMLProviderFromBytes([]byte(ptrYaml))
+
+	cfg := &pointerFieldStruct{Name: "xxx"}
+	v := p.Get("ps")
+
+	require.NoError(t, v.Populate(cfg))
+}
+
+func TestPointerTypedField(t *testing.T) {
+	t.Parallel()
+
+	type pointerFieldStruct struct {
+		Name  string
+		Value *int
+	}
+
+	var ptrPort = `
+ps:
+  name: Hello
+  port: 123
+`
+	p := NewYAMLProviderFromBytes([]byte(ptrPort))
+
+	cfg := &pointerFieldStruct{Name: "xxx"}
+	v := p.Get("ps")
+
+	require.NoError(t, v.Populate(cfg))
+}
+
+func TestPointerChildTypedField(t *testing.T) {
+	t.Parallel()
+
+	type Port int
+	type childPort struct {
+		Port *Port
```

Comment:
We already have plenty of them, e.g.: https://github.com/uber-go/fx/blob/dev/config/decoder_test.go#L233

This is a regression test from Shawn's PR.